lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+  <html>
+    <head><title><%= title %></title></head>
+    <body><%= content %></body>
+  </html>
+
+The stash or helpers like L<Mojolicious::Plugin::DefaultHelpers/"title"> can be used to pass additional data to the
+layout.
+
+=head2 Blocks
+
+Template blocks can be used like normal Perl functions and are always delimited by the C<begin> and C<end> keywords,
+they are the foundation for many helpers.
+
+  use Mojolicious::Lite;
+
+  get '/with_block' => 'block';
+
+  app->start;
+  __DATA__
+
+  @@ block.html.ep
+  % my $link = begin
+    % my ($url, $name) = @_;
+    Try <%= link_to $url => begin %><%= $name %><% end %>.
+  % end
+  <!DOCTYPE html>
+  <html>
+    <head><title>Sebastians frameworks</title></head>
+    <body>
+      %= $link->('http://mojolicious.org', 'Mojolicious')
+      %= $link->('http://mojojs.org', 'mojo.js')
+    </body>
+  </html>
+
+=head2 Helpers
+
+Helpers are little functions you can create with the keyword L<Mojolicious::Lite/"helper"> and reuse throughout your
+whole application, from actions to templates.
+
+  use Mojolicious::Lite -signatures;
+
+  # A helper to identify visitors
+  helper whois => sub ($c) {
+    my $agent = $c->req->headers->user_agent || 'Anonymous';
+    my $ip    = $c->tx->remote_address;
+    return "$agent ($ip)";
+  };
+
+  # Use helper in action and template
+  get '/secret' => sub ($c) {
+    my $user = $c->whois;
+    $c->app->log->debug("Request from $user");
+  };
+
+  app->start;
+  __DATA__
+
+  @@ secret.html.ep
+  We know who you are <%= whois %>.
+
+A list of all built-in ones can be found in L<Mojolicious::Plugin::DefaultHelpers> and
+L<Mojolicious::Plugin::TagHelpers>.
+
+=head2 Plugins
+
+Plugins are application extensions that help with code sharing and organization. You can load a plugin with the keyword
+L<Mojolicious::Lite/"plugin">, which can omit the C<Mojolicious::Plugin::> part of the name, and optionally provide
+configuration for the plugin.
+
+  use Mojolicious::Lite;
+
+  plugin Config => {file => '/etc/myapp.conf', default => {foo => 'bar'}};
+
+  # Return configured foo value, or default if no configuration file
+  get '/foo' => sub ($c) {
+    my $foo = $c->app->config('foo');
+    $c->render(json => {foo => $foo});
+  };
+
+  app->start;
+
+L<Mojolicious::Plugin::Config> is a built-in plugin which can populate L<Mojolicious/"config"> using a config file.
+Plugins can also set up routes, hooks, handlers, or even load other plugins. A list of built-in plugins can be found at
+L<Mojolicious::Plugins/"PLUGINS">, and many more are available from
+L<CPAN|https://metacpan.org/search?q=Mojolicious+Plugin>.
+
+=head2 Placeholders
+
+Route placeholders allow capturing parts of a request path until a C</> or C<.> separator occurs, similar to the
+regular expression C<([^/.]+)>. Results are accessible via L<Mojolicious::Controller/"stash"> and
+L<Mojolicious::Controller/"param">.
+
+  use Mojolicious::Lite -signatures;
+
+  # /foo/test
+  # /foo/test123
```

Comment:
It seems worth pointing out here that with `# /foo/test123?bar=test456`, `$c->stash('bar')` will only refer to the request path portion test123; whereas with `$c->param('bar')`, it could get it from the GET or POST data, too.